update jwt tokens lib

## Motivation

I have an idea how to improve the [quality](https://iso25000.com/index.php/en/iso-25000-standards/iso-25010) of the system. 

**Now, in current architecture**
`github.com/golang-jwt/jwt` is used

Problem - voulnerability detected: https://github.com/voedger/voedger/actions/runs/14099355519/job/39492776347
```
Vulnerability #1: GO-2025-3553
    Excessive memory allocation during header parsing in
    github.com/golang-jwt/jwt
  More info: https://pkg.go.dev/vuln/GO-2025-3553
  Module: github.com/golang-jwt/jwt
    Found in: github.com/golang-jwt/jwt@v3.2.2+incompatible
    Fixed in: N/A
    Example traces found:
Error:       #1: pkg/itokensjwt/impl.go:[20](https://github.com/voedger/voedger/actions/runs/14099355519/job/39492776347#step:4:21)2:31: itokensjwt.JWTSigner.sign calls jwt.NewWithClaims
Error:       #2: pkg/itokensjwt/impl.go:77:30: itokensjwt.JWTSigner.ValidateToken calls jwt.Parser.Parse
Error:       #3: pkg/itokensjwt/impl.go:204:31: itokensjwt.JWTSigner.sign calls jwt.Token.SignedString
Error:       #4: pkg/itokensjwt/impl.go:86:30: itokensjwt.JWTSigner.ValidateToken calls jwt.ValidationError.Error
Error:       #5: pkg/itokensjwt/impl.go:[22](https://github.com/voedger/voedger/actions/runs/14099355519/job/39492776347#step:4:23):2: itokensjwt.init calls jwt.init

Your code is affected by 1 vulnerability from 1 module.
This scan also found 0 vulnerabilities in packages you import and 2
vulnerabilities in modules you require, but your code doesn't appear to call
these vulnerabilities.
Use '-show verbose' for more details.
```

## Proposal

use `github.com/golang-jwt/jwt/v5`

--- a/go.mod
+++ b/go.mod
@@ -14,7 +14,7 @@ require (
 	github.com/emersion/go-sasl v0.0.0-20241020182733-b788ff22d5a6
 	github.com/emersion/go-smtp v0.21.3
 	github.com/gocql/gocql v1.7.0
-	github.com/golang-jwt/jwt v3.2.2+incompatible
+	github.com/golang-jwt/jwt/v5 v5.2.2
 	github.com/google/go-cmp v0.6.0
 	github.com/google/gofuzz v1.2.0
 	github.com/google/uuid v1.6.0
@@ -54,7 +54,6 @@ require (
 	github.com/aws/smithy-go v1.22.1 // indirect
 	github.com/cespare/xxhash/v2 v2.3.0 // indirect
 	github.com/davecgh/go-spew v1.1.1 // indirect
-	github.com/golang-jwt/jwt/v5 v5.2.2 // indirect
 	github.com/golang/snappy v0.0.4 // indirect
 	github.com/google/flatbuffers v24.12.23+incompatible // indirect
 	github.com/hailocab/go-hostpool v0.0.0-20160125115350-e80d13ce29ed // indirect

--- a/go.mod
+++ b/go.mod
@@ -54,6 +54,7 @@ require (
 	github.com/aws/smithy-go v1.22.1 // indirect
 	github.com/cespare/xxhash/v2 v2.3.0 // indirect
 	github.com/davecgh/go-spew v1.1.1 // indirect
+	github.com/golang-jwt/jwt/v5 v5.2.2 // indirect
 	github.com/golang/snappy v0.0.4 // indirect
 	github.com/google/flatbuffers v24.12.23+incompatible // indirect
 	github.com/hailocab/go-hostpool v0.0.0-20160125115350-e80d13ce29ed // indirect

--- a/pkg/itokensjwt/impl.go
+++ b/pkg/itokensjwt/impl.go
@@ -10,6 +10,7 @@ import (
 	"crypto/hmac"
 	"encoding/base64"
 	"encoding/json"
+	"errors"
 	"fmt"
 	"reflect"
 	"strings"
@@ -19,7 +20,7 @@ import (
 	"github.com/voedger/voedger/pkg/coreutils"
 	"github.com/voedger/voedger/pkg/istructs"
 
-	"github.com/golang-jwt/jwt"
+	"github.com/golang-jwt/jwt/v5"
 
 	"github.com/voedger/voedger/pkg/itokens"
 )
@@ -50,13 +51,14 @@ func (j *JWTSigner) IssueToken(app appdef.AppQName, duration time.Duration, poin
 		return "", err
 	}
 
+	now := j.iTime.Now()
 	claims := &jwt.MapClaims{
-		"iat":      jwt.TimeFunc().Unix(),
-		"exp":      jwt.TimeFunc().Add(duration).Unix(),
+		"iat":      now.Unix(),
+		"exp":      now.Add(duration).Unix(),
 		"aud":      audience.String(),
 		"Duration": duration,
 		"AppQName": &app,
-		"IssuedAt": jwt.TimeFunc(),
+		"IssuedAt": now,
 	}
 	*claims = mergeClaimsMaps(m, *claims)
 	token, err = j.sign(claims)
@@ -73,7 +75,7 @@ func (j *JWTSigner) ValidateToken(token string, pointerToPayload interface{}) (g
 		jwtToken *jwt.Token
 	)
 	expectedAudience := reflect.TypeOf(pointerToPayload).Elem().String()
-	parser := &jwt.Parser{UseJSONNumber: true}
+	parser := jwt.NewParser(jwt.WithJSONNumber(), jwt.WithTimeFunc(j.iTime.Now))
 	jwtToken, err = parser.Parse(token, func(token *jwt.Token) (interface{}, error) {
 		_, ok := token.Method.(*jwt.SigningMethodHMAC)
 		if !ok {
@@ -184,15 +186,11 @@ func (j *JWTSigner) CryptoHash256(data []byte) (hash [hashLength]byte) {
 
 //nolint:errorlint
 func setErrorDescription(err error) error {
-	if ve, ok := err.(*jwt.ValidationError); ok {
-		if ve.Errors&(jwt.ValidationErrorExpired) != 0 {
-			// Token is expired
-			err = itokens.ErrTokenExpired
-		}
-		if ve.Errors&(jwt.ValidationErrorUnverifiable) != 0 {
-			// Token malformed
-			err = itokens.ErrInvalidToken
-		}
+	if errors.Is(err, jwt.ErrTokenExpired) {
+		err = itokens.ErrTokenExpired
+	}
+	if errors.Is(err, jwt.ErrTokenUnverifiable) {
+		err = itokens.ErrInvalidToken
 	}
 	return err
 }
@@ -210,12 +208,12 @@ func getTokenPayload(token []string) string {
 	return token[1]
 }
 
-func NewJWTSigner(secretKey SecretKeyType) *JWTSigner {
+func NewJWTSigner(secretKey SecretKeyType, iTime coreutils.ITime) *JWTSigner {
 	var byteSecretKey []byte = secretKey
 	if len(byteSecretKey) < SecretKeyLength {
 		panic(fmt.Errorf("invalid key length: must be %d chars", SecretKeyLength))
 	}
-	return &JWTSigner{byteSecretKey}
+	return &JWTSigner{byteSecretKey, iTime}
 }
 
 func mergeClaimsMaps(maps ...map[string]interface{}) (result map[string]interface{}) {

--- a/pkg/itokensjwt/provide.go
+++ b/pkg/itokensjwt/provide.go
@@ -7,30 +7,14 @@
 package itokensjwt
 
 import (
-	"sync"
-
-	"github.com/golang-jwt/jwt"
 	"github.com/voedger/voedger/pkg/coreutils"
 	itokens "github.com/voedger/voedger/pkg/itokens"
 )
-
-// jwt.TimeFunc will be set to timeFunc but jwt.TimeFunc can not be protected from simulaneous access -> must be set only once
-// e.g. to avoid data race: write jwt.TimeFunc in next test, read in async istructs.Projector of a previous test
-var onceJWTTimeFuncSetter = sync.Once{}
-
-var initial coreutils.ITime
 
 // ProvideITokens implementation by provided interface
 // To receive implementation you must provide Secret Key. Min length - 64 byte, panic otherwise
 // panics on try to init again with different ITime implementation to avoid cases when we have >1 VIT that uses different times
 // jwt.TimeFunc is single point of the time per the whole process so all tests must share exactly one implementation of ITime
 func ProvideITokens(secretKey SecretKeyType, time coreutils.ITime) (tokenImpl itokens.ITokens) {
-	onceJWTTimeFuncSetter.Do(func() {
-		jwt.TimeFunc = time.Now
-		initial = time
-	})
-	if initial != time {
-		panic("tokens engine can not be initialized with different ITime implementation. Provide coreutils.MockTime only in tests and the singleton ITime implementation in runtime")
-	}
-	return NewJWTSigner(secretKey)
+	return NewJWTSigner(secretKey, time)
 }

--- a/pkg/itokensjwt/types.go
+++ b/pkg/itokensjwt/types.go
@@ -4,8 +4,11 @@
 
 package itokensjwt
 
+import "github.com/voedger/voedger/pkg/coreutils"
+
 type SecretKeyType []byte
 
 type JWTSigner struct {
 	secretKey []byte
+	iTime coreutils.ITime
 }

